mongodump/mongodump_test.go

```diff
@@ -872,6 +872,95 @@ func TestMongoDumpTOOLS2174(t *testing.T) {
 	})
 }
 
+// Test dumping a collection while respecting no index scan for wired tiger.
+func TestMongoDumpTOOLS1952(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	log.SetWriter(ioutil.Discard)
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("Failed to get session: %v", err)
+	}
+
+	collName := "tools-1952"
+	dbName := "test"
+
+	var r1 bson.M
+
+	dbStruct := session.Database(dbName)
+
+	sessionProvider.Run(bson.D{{"drop", collName}}, &r1, dbName)
+
+	createCmd := bson.D{
+		{"create", collName},
+	}
+	var r2 bson.M
+	err = sessionProvider.Run(createCmd, &r2, dbName)
+	if err != nil {
+		t.Fatalf("Error creating capped, no-autoIndexId collection: %v", err)
+	}
+
+	// Check whether we are using WiredTiger.
+	isWiredTiger := db.IsWiredTiger(dbStruct, collName)
+
+	// Turn on profiling.
+	profileCmd := bson.D{
+		{"profile", 2},
+	}
+
+	err = sessionProvider.Run(profileCmd, &r2, dbName)
+	if err != nil {
+		t.Fatalf("Failed to turn on profiling: %v", err)
+	}
+
+	profileCollection := dbStruct.Collection("system.profile")
+
+	Convey("testing dumping a capped, autoIndexId:false collection", t, func() {
+		md := simpleMongoDumpInstance()
+		md.ToolOptions.Namespace.Collection = collName
+		md.ToolOptions.Namespace.DB = dbName
+		md.OutputOptions.Out = "dump"
+		err = md.Init()
+		So(err, ShouldBeNil)
+		err = md.Dump()
+		So(err, ShouldBeNil)
+
+		if !isWiredTiger {
```

Comment:
right now, it looks like we're not asserting anything when the storage engine _is_ wiredTiger. Without such an assertion, I'm not sure this test is actually verifying the behavior added in this ticket.

Reply:
There is nothing to assert.

Reply:
Well, $natural gives us something to assert :)